danimart1991/visual-studio-team-services-desde-el-terminal/readme.md

```diff
@@ -0,0 +1,136 @@
+
+Para ver el listado de las *Pull Request* en las que trabaja el equipo en los diferentes proyectos basta con acceder a la sección ``https://{account}.visualStudio.com/_pulls``. El problema con esta página es que no posee notificaciones, por tanto, pese a ser una buena práctica acceder a la página nada más empezar la jornada laboral y revisar las *Pull Request* pendientes de tu equipo, es cierto que en una jornada de 8 horas pueden surgir una gran cantidad de *Pull Request* y al final acabas teniendo siempre la página abierta.
+
+Lo que hacemos hasta el momento es tener notificaciones con algunas alertas a través de correo electrónico, que es actualmente lo que permite *Microsoft* en su herramienta. Sin embargo, que os podemos contar de los correos electrónicos, en la mayoría de casos tenemos una regla de que todos los avisos que suceden en *VSTS (Visual Studio Team Services)* se pasen a una carpeta silenciada que se revisa cada varios días porque no son avisos tan importantes como para tenerlos en cuenta.
+
+Como he dicho usamos *Git*, y además como somos puristas solemos usar la consola para realizar comandos sobre el repositorio. ¿Y si no necesitásemos tener abierto el correo electrónico, ni el navegador web para realizar la consulta de las *Pull Request* pendientes del equipo? Entra en juego [Microsoft Visual Studio Team Services CLI](https://github.com/Microsoft/vsts-cli).
+
+> **[VSTS CLI](https://docs.microsoft.com/en-us/cli/vsts/overview)** is a new command line interface for *Visual Studio Team Services (VSTS)* and *Team Foundation Server (TFS) 2017 Update 2* and later.
+
+En que se traduce esto. Pues a grandes rasgos es una interfaz para nuestra línea de comandos que agrega funcionalidades extra para trabajar con *Visual Studio Team Services* y *Team Foundation Server*.
+
+### Instalación
+
+**VSTS CLI** tiene soporte **Cross-Platform** para *Windows, Linux, MacOS* e imagen *Docker*, y para instalarlo solo hemos de seguir este [**link**](https://docs.microsoft.com/es-es/cli/vsts/install?view=vsts-cli-latest). En *Windows* la instalación es tan sencilla como descargar y ejecutar un instalador.
+
+Una vez instalado, podemos abrir nuestra consola favorita y ejecutar los siguientes comandos. Si recibimos un listado de comandos disponibles la instalación ha sido exitosa:
+
+```bash
+vsts -h
+vsts code -h
+```
+
+### Iniciando sesión
+
+Para poder acceder a la información contenida en la instancia de *Visual Studio Team Services* en la que solemos trabajar necesitamos iniciar sesión, para ello ejecutamos el siguiente comando:
+
+``vsts login --instance https://MYACCOUNT.visualstudio.com --token MYTOKEN``
+
+Para obtener el *token* necesario, basta con acceder a la sección seguridad de nuestra cuenta en la instancia de **VSTS** y solicitar un **Personal Access Token**. [Más información aquí](https://docs.microsoft.com/es-es/vsts/accounts/use-personal-access-tokens-to-authenticate).
```

Comment:
Más información [aquí](https://docs.microsoft.com/es-es/vsts/accounts/use-personal-access-tokens-to-authenticate).